packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -990,10 +991,8 @@ export async function getBase(
 	const webhookWaitingBaseUrl = urlBaseWebhook + config.getEnv('endpoints.webhookWaiting');
 	const webhookTestBaseUrl = urlBaseWebhook + config.getEnv('endpoints.webhookTest');
 
-	const [encryptionKey, variables] = await Promise.all([
-		UserSettings.getEncryptionKey(),
-		WorkflowHelpers.getVariables(),
-	]);
+	const encryptionKey = await UserSettings.getEncryptionKey();
+	const variables = isVariablesEnabled() ? await WorkflowHelpers.getVariables() : {};
```

Comment:
Would it make sense to move this check to inside the helper rather than at this level? Asking because if I wanted ot fetch all variables I'd go to the helper directly, so it's a higher chance I'd get the correct logic there.